web/tests/calculate-percentage-survey-spec.js

```diff
@@ -0,0 +1,68 @@
+describe('FieldsetParserService', function () {
+    const questionSet = {
+        1 : {
+            fieldGroup : [
+                1,
+                2,
+                3
+            ]
+        },
+        2 : {
+            fieldGroup : [
+                1,
+                2,
+                3
+            ]
+        }
+    };
+    const answerSet = {
+        1: {
+            2: 9,
+            3: 0
+        },
+        2: {
+            4: 0,
+            5: 6,
+        }
+    }
+
+    beforeEach(function () {
+        angular.mock.module('incuna-surveys.calculate-completion-percent');
+
+        inject(function (calculateCompletionPercent) {
+            this.Parser = calculateCompletionPercent;
+        });
+
+        this.totalQuestions = this.Parser.countQuestionsTotal(questionSet);
+        this.totalQuestionsAnswered = this.Parser.countNumberOfAnsweredQuestions(answerSet);
+    });
+
+    describe('countQuestionsTotal function', function () {
+        it('should return the total number of questions', function () {
+            expect(this.totalQuestions).toBe(6);
+        });
+    });
+
+    describe('countNumberOfAnsweredQuestions function', function () {
+        it('should return the total number of questions answered', function () {
+            expect(this.totalQuestionsAnswered).toBe(2);
+        });
+    });
+
+    describe('calculatePercentageComplete function', function () {
+        it('should return the 0 percentage when none answered', function () {
+            this.percentage = this.Parser.calculatePercentageComplete(0, this.totalQuestions);
```

Comment:
You don;t need `this` here as these are local variavbles. Can be just `const percentage`